### ISSUE TITLE
Feature request: link to notes

### ISSUE THREAD

**Issue by terrychenzw** [clairefro/obsidian-plugin-chronos#7]
I really like this awesome plugin! thanks for your great effort on creating this plugin.

Is it possible to allow adding links to the event block?  When clicking an event block, it is able to open the target note.

**Comment by Elaws** [comment]
That would be amazing, was going to publish that same feature request !

**Comment by clairefro** [comment]
Thanks for the input! I'm planning to make Chronos available outside Obsidian so your timelines can live anywhere, so let me think about how to approach this

**Comment by clairefro** [comment]
update - I have note linking working in a test branch, but having issues with updating links when you rename or move a note. stay tuned

**Comment by Elaws** [comment]
That's some great news, thank you !

**Comment by slartybartfastest** [comment]
So glad to see this as a WIP!
Thanks for your rigorous testing. This is the feature I most wanted added after testing your plug-in. So happy to come here and see it already underway.

**Comment by TonyAtlas** [comment]
I also thank you for working on this option. It will make the plugin much more useful.

**Comment by clairefro** [comment]
Thanks for your feedback (and patience!) I started a new job this week and have a commitment all weekend, so will likely be next week

**Comment by Brett-ai** [comment]
Just came here to say this plugin looks almost perfect for creating timelines in Obsidian, but the one thing that would make it perfect is the ability to link to notes! I see others have beaten me to it :)

**Comment by RONNYKHALIL** [comment]
+1 - just found this plugin after trying way too many other options to make simple navigable timelines in obsidian. chronos is by far my fav (great name too!) and the ability to link to notes would seal the deal. thank you for this contribution!!!

**Comment by clairefro** [comment]
@Elaws great suggestion! I will try

**Comment by clairefro** [comment]
Update on this - everything is working fine but still having trouble with updating filepaths when you rename or move a linked note. 

... Somehow in testing I overwrote all the obsidian files in my personal drive with the word 'foo'. It goes without saying it's not ready to ship 😆 and I need to address recovering my personal files

**Comment by Elaws** [comment]
Thank you for the update ! I hope you get your files back soon =D

**Comment by clairefro** [comment]
Note linking is now available in [1.1.0](https://github.com/clairefro/obsidian-plugin-chronos/releases/tag/1.1.0)

Due to technical obstacles, only links to notes with a full path are automatically updated when renamed/moved. Will work on getting aliased notes to update safely.

**Comment by clairefro** [comment]
````markdown
```chronos
- [2021~2022] No link
- [2023~2024] With link [[path/to/note]]
- [2022~2024] Link in description | [[path/to/note]]
* [2022] Link in description | [[path/to/note]]
```
````
![ex-note-linking-2](https://github.com/user-attachments/assets/b0912f93-e5bf-4832-931a-6ef792922cec)

**Comment by Elaws** [comment]
> Note linking is now available in [1.1.0](https://github.com/clairefro/obsidian-plugin-chronos/releases/tag/1.1.0)
> 
> Due to technical obstacles, only links to notes with a full path are automatically updated when renamed/moved. Will work on getting aliased notes to update safely.

It's fantastic, many thanks !

A few remarks after testing it a bit (if you wish, I can open separate bug report or FR for each) : 

- Attachement in my notes use the note name with incrementing numbers (e.g. : note name is `Note.md`, attachments are `Note.png`, `Note 1.png`, etc...). Using `[[Note]]` in a chronos block will redirect to `Note.png` and not `Note.md`. If that note is not in the vault root, `path/to/Note` will properly open `Note.md` and not `Note.png`.

- Middle click to open the link in a new tab does not work, it could be useful. Maj + click does work properly though.

- When clicking on a link, the color changes. Is this the expected behavior ? Reloading the block loads the previous color.

![image](https://github.com/user-attachments/assets/9abaabb8-a750-4ce7-8e96-cf48bdf90852)

- I thought we could hide the `[[]]` in the link (either in links or links in description) : what do you think ?

Anyway, amazing plugin, even more with the links. Looking forward to seeing the future improvements !

**Comment by clairefro** [comment]
Thanks for the input! 
1.  and 2. are enlightening - will open it as new issues!

3. is expected actually - you'll see the some thing happen on a non-linked note. It came out of the box like that and I decided to keep it because it can be useful if you have tons of items on a page and want to see the last one you clicked before switching tabs, or want to highlight one temporarily. I'm also considering adding two-way data binding you can edit the markdown by clicking around in the GUI, so planning to add interactivity there and didn't bother overwriting the native select logic 

4. I went back and forth on it before release, and decided to keep the `[[]]` to clarify they are links, no matter where they show. I certainly think it's ugly in the event title, and personally rather stow links in the description and call the event whatever I want with the (similar to `(title text)[this/is/a/link]` in markdown ). I think it helps to see the `[[]]` indicating a link in the description so notes with aliases are clearly separate from description text:

````markdown
```chronos
- [1789] French Revolution| Period of radical political and social upheaval in France, leading to the overthrow of the monarchy, the rise of democracy and secularism, and significant societal transformations influenced by Enlightenment ideals [[note]]
```
````

However, I'm very keen to hear other opinons on this

**Comment by Elaws** [comment]
Thank you !

Concerning point n°4, how about rendering [[link]] as <ins>link</ins> ?

So :

- `[2023~2024] With link [[path/to/note]]` would be rendered as "With link <ins>path/to/note</ins>"
- `[2022~2024] Link in description | [[path/to/note]]` as it is already, with `path/to/note` underlined in the description.

**Comment by clairefro** [comment]
@Elaws that was actually my initial idea! But then I realized it wasn't clear the item is clickable if the link is hidden in the the description.

One idea is to use a different link signal, like a border around items with links (then could underline links like your suggestion above). Will play with it

**Comment by clairefro** [comment]
PS @Elaws - 2) has been addressed in release [1.1.1](https://github.com/clairefro/obsidian-plugin-chronos/releases/tag/1.1.1) - left-click opens link in current tab, middle-click/shift +click opens in new tab

**Comment by Elaws** [comment]
@clairefro : Thank you ! I just tested that feature, and I encountered a few issues : 

1/ If you middle-click on a link, the note does open on a new tab, but the mouse cursor is stuck to the "scroll mode" icon :

![image](https://github.com/user-attachments/assets/c063db75-909a-4122-a5e6-5ec041270953)

Note : it does not seem to trigger scroll mode, it only shows the scroll mode icon.

2/ If you click on a link without target (e.g. : note not created yet), mouse switches to scroll mode.

I can open a new issue if needed !
